### ISSUE TITLE
Time progression syntax issue, run mode issue, etc

### ISSUE THREAD

**Issue by vincentjames501** [synapse-wireless-labs/jasmine-marbles#30]
I was playing around trying to get the time progression syntax working but began to run into quite a few hurdles.

First, the time progression syntax only works in the context of:

```javascript
getTestScheduler().run(() => {...})
```

If you look at the source code of `TestScheduler.parseMarbles` it needs to be in `runMode` for time progression syntax to be parsed.

My next thought is OK, I'll just wrap my code in `.run`. Turns out you can't use the `toBeObservable` matcher in inside of `.run` because https://github.com/synapse-wireless-labs/jasmine-marbles/blob/master/index.ts#L145 reparses the marbles and loses the context of `runMode` (note the last argument to `TestScheduler.parseMarbles` is `runMode`).

I think some careful consideration should be taken as some things should likely be redesigned to better work with `.run` for its advantages.

**Comment by damienwebdev** [comment]
@MikeRyanDev @brandonroberts this is pretty frustrating. I spent a fair chunk of time wondering what I was doing wrong.

I followed the [rxjs docs](https://github.com/ReactiveX/rxjs/blob/master/doc/marble-testing.md#Time-progression-syntax) to the letter only to find out that this never really worked as anticipated.

As a trivial example, consider the following code:

```ts
import { TestBed } from '@angular/core/testing';

import { Observable } from 'rxjs';

import { Action } from '@ngrx/store';
import { ROUTER_NAVIGATION } from '@ngrx/router-store';
import { provideMockActions } from '@ngrx/effects/testing';
import { Actions } from '@ngrx/effects';

import { hot, cold, getTestScheduler } from 'jasmine-marbles';

import { SidebarEffects } from './sidebar.effects';
import * as SidebarActions from '../actions/sidebar.actions';

describe('Daffio | SidebarEffects', () => {
  let effects: SidebarEffects;
  let actions$: Observable<any>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      providers: [
        SidebarEffects,
        provideMockActions(() => actions$)
      ]
    });

    effects = TestBed.get(SidebarEffects);
  });

  it('should be created', () => {
    expect(effects).toBeTruthy();
  });

  fdescribe('when a ROUTER_NAVIGATION occurs', () => {
    const action: Action = { type: ROUTER_NAVIGATION};
    const completion = new SidebarActions.CloseSidebar();

    it('should dispatch CloseSidebar with a 18ms delay', () => {
        let expected; 

        actions$ = hot( '--a', { a: action });
        expected = cold('--- 8ms b', { b: completion });

        expect(effects.closeOnPageChange$(18, getTestScheduler())).toBeObservable(expected);
    });
  });
});
```

It seems impossible for me to test anything with non-multiple of `10` passage of time. Additionally, it appears that you're parsing ` ` characters as passage of time as well. Is there a plan to address this issue?

**Comment by bbonczek** [comment]
Is this issue going to be addressed?

**Comment by craigsnyders** [comment]
I would expect this test to pass, but unfortunately the jasmine-marbles syntax fails. Regular Rxjs syntax passed (using expectObservable).

```
  it('should work', () => {
    const scheduler = getTestScheduler();
    scheduler.run((helpers) => {
      const { expectObservable } = helpers;

      const o = interval(1000).pipe(take(3), map((i) => `foo_${i}`));

      expectObservable(o).toBe('1s 0 999ms 1 999ms (2|)', ['foo_0', 'foo_1', 'foo_2']);
      expect(o).toBeObservable(hot('1s 0 999ms 1 999ms (2|)', ['foo_0', 'foo_1', 'foo_2']));
    });
  });
```

The weird passage of time (only-multiples-of-10) is also an annoyance, but my most pressing issue is that testing `interval` etc is not possible using the jasmine-marbles syntax. It doesn't even seem to parse it because each string ('100ms') gets interpreted literally.

**Comment by brandonroberts** [comment]
Support for time progression has been published in `0.5.0`
